src/lib/core/TLVReader.cpp

```diff
@@ -324,6 +328,49 @@ CHIP_ERROR TLVReader::Get(CharSpan & v)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR TLVReader::Get(Optional<LocalizedStringIdentifier> & lsid)
+{
+    lsid.ClearValue();
+    VerifyOrReturnError(TLVTypeIsUTF8String(ElementType()), CHIP_ERROR_WRONG_TLV_TYPE);
+
+    const uint8_t * bytes;
+    ReturnErrorOnFailure(GetDataPtr(bytes)); // Does length sanity checks
+
+    uint32_t len = GetLength();
+
+    const uint8_t * infoSeparator1 = reinterpret_cast<const uint8_t *>(memchr(bytes, kUnicodeInformationSeparator1, len));
```

Comment:
I don't know too much about Unicode, but I'm not sure finding 0x1F in the byte array indicates that it's the start of a code point.

Reply:
According to the spec this special symbol (0x1F) should be uniquely distigutable:
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/6158/files#diff-05653d18812836190adf870ff988791e0e3901478a63ab1eca4c3e9df0168a8cR2378

Reply:
I was referring to code points where byte 2, byte 3, byte 4 might contain 0x1F. I thought the spec specified Unicode, but I see it specifies UTF-8 encoding. UTF-8 encodings byte 2 to byte 4 must be 0b10xxxxxx, so it is impossible for byte 2 to byte 4 to be 0x1F. P.S. I did find UTF-16 characters whose non-first byte is 0x1F.

Reply:
Your observation is correct. In UTF-8 encoding the collision of bytes 2 to 4 (when present) with 0x1F is impossible.